lib/wootric/test.js

```diff
@@ -0,0 +1,155 @@
+
+var Analytics = require('analytics.js').constructor;
+var integration = require('analytics.js-integration');
+var plugin = require('./');
+var tester = require('analytics.js-integration-tester');
+var sandbox = require('clear-env');
+
+describe('Wootric', function(){
+  var Wootric = plugin;
+  var wootric;
+  var analytics;
+  var options = {
+    account_token: 'NPS-01fe3cbc'
+  };
+
+  beforeEach(function(){
+    analytics = new Analytics;
+    wootric = new Wootric(options);
+    analytics.use(plugin);
+    analytics.use(tester);
+    analytics.add(wootric);
+    //setup Wootric to survey immediately
+    window.wootric_survey_immediately = true;
+  });
+
+  afterEach(function(){
+    analytics.restore();
+    analytics.reset();
+    wootric.reset();
+    sandbox();
+  });
+
+  it('should have the right settings', function(){
+    analytics.compare(Wootric, integration('Wootric')
+      .assumesPageview()
+        .option('account_token', '')
+        .global('wootricSettings')
+        .global('wootric_survey_immediately')
+        .global('wootric')
+    )
+  });
+
+  describe('before loading', function(){
+    beforeEach(function(){
+      analytics.stub(wootric, 'load');
+    });
+
+    it('should not have a wootric object', function(){
+      analytics.assert(!window.wootric);
+    })
+
+    describe('#initialize', function(){
+      beforeEach(function(){
+        analytics.initialize();
+        analytics.page();
+      });
+
+      it ('should setup the wootricSettings object', function(){
+        analytics.assert(window.wootricSettings);
+        analytics.assert(window.wootricSettings instanceof Object)
+      })
+
+      it('should call #load', function(){
+        analytics.called(wootric.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(wootric, done);
+    });
+  });
+
+  describe('after loading', function(){
```

Comment:
We should stub out `wootric.load` and any necessary Wootric functions so these tests aren't dependent on loading the Wootric library onto the page. That way we're testing the integration, not the Wootric library.

For a good example of how to go about this, check out the Clicktale and Inspectlet tests:

https://github.com/segmentio/analytics.js-integrations/blob/3f103e6d7d7c3cb2dc299c5bdfdf17532c92638c/lib/inspectlet/test.js
https://github.com/segmentio/analytics.js-integrations/blob/98bc2b34847faed0cd3d319687087e45f49bc2b6/lib/clicktale/test.js